### PR TITLE
Add worker deployments for spackbotdev

### PR DESCRIPTION
**Pull request by kwryankrattiger**
@scottwittenburg

### CODE REVIEW

**Comment by kwryankrattiger** [comment]
@zackgalbreath I moved the close stuff to spackbot since I am also deleting the mirror there https://github.com/spack/spackbot/pull/68

Could you review those changes to make sure I didn't break something else as well.

File: images/gh-gl-sync/SpackCIBridge.py [hunk 1]
```diff
@@ -601,13 +558,6 @@ def sync(self):
             # Get tags on GitHub.
             tags = self.list_github_tags()
 
-            # Retrieve PRs that have already been synced to GitLab.
-            synced_prs = self.get_synced_prs()
-
-            # Find closed PRs that are currently synced.
-            # These will be deleted from GitLab.
-            closed_refspecs = self.get_prs_to_delete(all_open_prs["pr_strings"], synced_prs)
```
Comment (zackgalbreath) [inline images/gh-gl-sync/SpackCIBridge.py:609]: @kwryankrattiger can you show me the new spackbot code that deletes branches from GitLab when a PR is merged on GitHub?

**Review by zackgalbreath — CHANGES_REQUESTED** [review]
You will also need to bump the version number for gh-gl-sync in .github/workflows/custom_docker_builds.yml and k8s/custom/gh-gl-sync/cron-jobs.yaml

File: k8s/spack/spackbotdev-spack-io/deployments.yaml [hunk 2]
```diff
@@ -95,27 +101,35 @@ spec:
...
         resources:
           requests:
             cpu: 900m
             memory: 1G
         # Mount secrets to non-existing location
         volumeMounts:
-          - mountPath: "/git_rsa"
-            name: spack-bot-dev-idrsa
-            readOnly: true
+        - mountPath: "/git_rsa"
```
Comment (scottwittenburg) [inline k8s/spack/spackbotdev-spack-io/deployments.yaml:114]: I'm not sure whether indented or not indented is preferred here.  But somehow I managed to end up with the indentation not matching between the production and development versions.  Can you either add indents here, or remove them in the dev version?

File: k8s/spack/spackbotdev-spack-io/deployments.yaml [hunk 3]
```diff
@@ -145,10 +159,97 @@ spec:
...
+            secretKeyRef:
+              name: spack-bot-dev-credentials
+              key: github_private_key
+        - name: GITHUB_WEBHOOK_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: spack-bot-dev-credentials
+              key: github_webhook_secret
+      volumes:
+      - name: spack-bot-dev-idrsa
```
Comment (scottwittenburg) [inline k8s/spack/spackbotdev-spack-io/deployments.yaml:249]: indentation match between production/dev

File: k8s/spack/spackbotdev-spack-io/deployments.yaml [hunk 4]
```diff
@@ -145,10 +159,97 @@ spec:
...
+        # Can use some other image for testing, see README.md
+        image: "ghcr.io/kwryankrattiger/spackbot-workers:0.0.1"
+        imagePullPolicy: Always
+        resources:
+          requests:
+            cpu: 900m
+            memory: 1G
+        # Mount secrets to non-existing location
+        volumeMounts:
+        - mountPath: "/git_rsa"
```
Comment (scottwittenburg) [inline k8s/spack/spackbotdev-spack-io/deployments.yaml:203]: indentation match between production/dev

File: k8s/spack/spackbotdev-spack-io/deployments.yaml [hunk 5]
```diff
@@ -145,10 +159,97 @@ spec:
               name: spack-bot-dev-credentials
               key: github_webhook_secret
       volumes:
-        - name: spack-bot-dev-idrsa
-          secret:
-            secretName: spack-bot-dev-idrsa
-            defaultMode: 0600
+      - name: spack-bot-dev-idrsa
```
Comment (scottwittenburg) [inline k8s/spack/spackbotdev-spack-io/deployments.yaml:162]: indentation match between production/dev

**Review by scottwittenburg — CHANGES_REQUESTED** [review]
This is looking pretty good @kwryankrattiger thanks!  Couple requests:

I feel like this is not rebased on the latest `main`, to make it easier for me to make another review pass, can you rebase it?

Somehow the nested contents of the `volumes` and `volumeMounts` sections in the deployments are indented differently between spackbot-spack-io and spackbotdev-spack-io.  Can you make them match?

Thanks!

File: images/gh-gl-sync/SpackCIBridge.py (hunk 1 above)
Comment (kwryankrattiger) [reply images/gh-gl-sync/SpackCIBridge.py:609]: Here is the line the operation is handled:

https://github.com/spack/spackbot/pull/68/files#diff-b8354323f0494dadd3ec1c8a317ce0f3f685d3ca78be54403f325d1ad0101580R61

And here is where it is called.

https://github.com/spack/spackbot/pull/68/files#diff-9afe3332bf5162c85164c3a680f549ee6c26428ffded12cf84de73e93ccce62eR142

**Review by scottwittenburg — APPROVED** [review]

**Review by zackgalbreath — APPROVED** [review]

**Pull request merged**